pdata/plog/generated_scopelogs.go

```diff
@@ -28,34 +28,60 @@ import (
 // This is a reference type, if passed by value and callee modifies it the
 // caller will see the modification.
 //
-// Must use NewScopeLogs function to create new instances.
+// Must use NewExemplar function to create new instances.
 // Important: zero-initialized instance is not valid for use.
 type ScopeLogs struct {
+	commonScopeLogs
+}
+
+type MutableScopeLogs struct {
+	commonScopeLogs
+	preventConversion struct{}
+}
+
+type commonScopeLogs struct {
 	orig *otlplogs.ScopeLogs
 }
 
-func newScopeLogs(orig *otlplogs.ScopeLogs) ScopeLogs {
-	return ScopeLogs{orig}
+func newImmutableScopeLogs(orig *otlplogs.ScopeLogs) ScopeLogs {
```

Comment:
s/newImmutableScopeLogs/newScopeLogs

Reply:
We need to export the same helper for the common structures in the internal package where we would have `NewImmutableInstrumentationScope(orig)` -> `NewInstrumentationScope(orig)` that will conflict with `NewInstrumentationScope()`. So I kept the names consistent across pdata both for common structs (exported) and not exported.

Reply:
But it is not consistent: func new**Immutable**ScopeLogs(orig *otlplogs.ScopeLogs) ScopeLogs -> here the Immutable is extra....

Reply:
I know. I added the redundant `Immutable` part of all the structs. We can remove in for this struct but cannot remove in from:

```
func NewImmutableInstrumentationScope(orig *otlpcommon.InstrumentationScope) InstrumentationScope
```

because it'll conflict with

```
func NewInstrumentationScope() InstrumentationScope
```

Reply:
Discussed offline, resolved